internal/api/grpc/group/v2/integration_test/group_users_test.go

```diff
@@ -0,0 +1,336 @@
+		},
+		{
+			name: "missing permission, error",
+			ctx:  instance.WithAuthorizationToken(CTX, integration.UserTypeNoPermission),
+			req: &group_v2.AddUsersToGroupRequest{
+				Id:      group.GetId(),
+				UserIds: []string{user1.GetUserId(), user2.GetUserId()},
+			},
+			wantErrCode: codes.NotFound,
+			wantErrMsg:  "membership not found (AUTHZ-cdgFk)",
+		},
+		{
+			name: "organization owner, missing permission, error",
+			ctx:  instance.WithAuthorizationToken(CTX, integration.UserTypeOrgOwner),
+			req: &group_v2.AddUsersToGroupRequest{
+				Id:      group.GetId(),
+				UserIds: []string{user1.GetUserId(), user2.GetUserId()},
+			},
+			wantErrCode: codes.NotFound,
+			wantErrMsg:  "membership not found (AUTHZ-cdgFk)",
+		},
+		{
+			name: "organization owner, with permission, ok",
+			ctx:  instance.WithAuthorizationToken(CTX, integration.UserTypeOrgOwner),
+			req: &group_v2.AddUsersToGroupRequest{
+				Id:      defOrgGroup.GetId(),
+				UserIds: []string{defOrgUser.GetUserId()},
+			},
+		},
+		{
+			name: "some users already in the group, ok",
+			ctx:  iamOwnerCtx,
+			req: &group_v2.AddUsersToGroupRequest{
+				Id:      group.GetId(),
+				UserIds: []string{user1.GetUserId(), user2.GetUserId(), user3.GetUserId()},
+			},
+			wantChangeDate: true,
+		},
+		{
+			name: "some users not found, ok",
+			ctx:  iamOwnerCtx,
+			req: &group_v2.AddUsersToGroupRequest{
+				Id:      group.GetId(),
+				UserIds: []string{user1.GetUserId(), user2.GetUserId(), "randomUser"},
+			},
+			wantErrCode: codes.FailedPrecondition,
+			wantErrMsg:  "User could not be found (COMMAND-uXHNj)",
+		},
+		{
+			name: "user in a different org not added, ok",
+			ctx:  iamOwnerCtx,
+			req: &group_v2.AddUsersToGroupRequest{
+				Id:      group.GetId(),
+				UserIds: []string{user1.GetUserId(), user2.GetUserId(), defOrgUser.GetUserId()},
+			},
+			wantErrCode: codes.FailedPrecondition,
+			wantErrMsg:  "User could not be found (COMMAND-uXHNj)",
+		},
+		{
+			name: "add all users to group, ok",
+			ctx:  iamOwnerCtx,
+			req: &group_v2.AddUsersToGroupRequest{
+				Id:      group.GetId(),
+				UserIds: []string{user1.GetUserId(), user2.GetUserId()},
+			},
+			wantChangeDate: true,
+		},
+		{
+			name: "add all users to group (with duplicate user IDs), ok",
+			ctx:  iamOwnerCtx,
+			req: &group_v2.AddUsersToGroupRequest{
+				Id:      group.GetId(),
+				UserIds: []string{user1.GetUserId(), user4.GetUserId(), user4.GetUserId()},
+			},
+			wantChangeDate: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := instance.Client.GroupV2.AddUsersToGroup(tt.ctx, tt.req)
+			if tt.wantErrCode != codes.OK {
+				require.Error(t, err)
+				require.Empty(t, got.GetChangeDate())
+				assert.Equal(t, tt.wantErrCode, status.Code(err))
+				assert.Equal(t, tt.wantErrMsg, status.Convert(err).Message())
+				return
+			}
+			require.NoError(t, err)
+			if tt.wantChangeDate {
+				require.NotEmpty(t, got.GetChangeDate())
```

Comment:
You can check if the change date is between the time before and after the APi call.